huaweicloud/services/ddm/data_source_huaweicloud_ddm_instances.go

```diff
@@ -0,0 +1,187 @@
+// ---------------------------------------------------------------
+// *** AUTO GENERATED CODE ***
+// @Product DDM
+// ---------------------------------------------------------------
+
+package ddm
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk/pagination"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func DataSourceDdmInstances() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: resourceDdmInstancesRead,
+		Schema: map[string]*schema.Schema{
+			"region": {
```

Comment:
I think we shoud support more query options for the data source, for example: name, status, enterprise_project_id

Reply:
done